simulation/behavior_tree_plugin/include/behavior_tree_plugin/transition_events/reset_request_event.hpp

```diff
@@ -0,0 +1,43 @@
+// Copyright 2015-2020 Tier IV, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef BEHAVIOR_TREE_PLUGIN__TRANSITION_EVENTS__RESET_REQUEST_EVENT_HPP_
+#define BEHAVIOR_TREE_PLUGIN__TRANSITION_EVENTS__RESET_REQUEST_EVENT_HPP_
+
+#include <behavior_tree_plugin/transition_events/transition_event.hpp>
+#include <memory>
+#include <rclcpp/rclcpp.hpp>
+
+namespace behavior_tree_plugin
+{
+class ResetRequestEvent : public TransitionEvent
+{
+public:
+  ResetRequestEvent(
+    const std::shared_ptr<BT::TreeNode> & root_node,
+    std::function<std::string()> get_request_function,
+    std::function<void(std::string)> set_request_function);
+  const std::string getCurrentAction() const;
+
+private:
+  void callback(
+    BT::Duration timestamp, const BT::TreeNode & node, BT::NodeStatus prev_status,
+    BT::NodeStatus status) override;
+  const std::shared_ptr<BT::TreeNode> & root_node_;
```

Comment:
I have a reference type as a member, is that what you intended?
It can be a dangling reference, so it seems dangerous.

Reply:
fixed in https://github.com/tier4/scenario_simulator_v2/pull/570/commits/5491ef1908e42f0a0fd4de872834eebfbed9e72a